Feature/pers 151

MiVvlt opened the pull request.

MiVvlt committed:
queries

MiVvlt committed:
promise fixes and refactoring

MiVvlt committed:
getCopySourceQuery

MiVvlt committed:
temporary fixes

MiVvlt committed:
fix

MiVvlt committed:
refactoring predicates functionality with escape function

MiVvlt committed:
refactored getPressReleaseAttachmentInsertQueries

MiVvlt committed:
fixes readme

MiVvlt committed:
fixes

MiVvlt committed:
refactored getPressReleaseRelationsInsertQuery

MiVvlt committed:
comment

MiVvlt committed:
renamed file

MiVvlt committed:
config for sources

MiVvlt committed:
added status relation

MiVvlt committed:
copy collaborationEvent

MiVvlt committed:
fixes

MiVvlt committed:
readme and cleanup

erikap commented on README.md:
````diff
@@ -22,8 +23,13 @@ services:
 
+```elixir
+post "/collaboration-activities/:id/share", @json do
+    forward conn, [], "http://press-releases-collaboration-service/collaboration-activities/" <> id <> "/share"
````
In de `docker-compose.yml` zou ik deze service gewoon `collaboration` noemen (dus ook de `docker-compose.yml` snippet in README updaten)

erikap commented on README.md:
```diff
@@ -34,10 +40,14 @@ and add the following route to the
 |-------|-------------|
 | 202 | Accepted |
-|403| Forbidden: indien de request niet uitgevoerd wordt door een gebruiker die deel uitmaakt van het master-kabinet dat het persbericht aangemaakt heeft (af te leiden uit de session-uri in de request headers)|
+|403 | Forbidden: indien de request niet uitgevoerd wordt door een gebruiker die deel uitmaakt van het master-kabinet dat het persbericht aangemaakt heeft (af te leiden uit de session-uri in de request headers)|
```
Nog een korte beschrijving van wat het endpoint doet toevoegen

erikap commented on sparql-helpers/collaboration-activities.sparql.js:
```diff
@@ -0,0 +1,58 @@
+    const queryResult = await query(`
+    ${PREFIXES}
+    
+    SELECT ?collaborationActivityURI ?pressReleaseURI ?startedAtTime ?editor ?editorId
```
Hier worden variables geselecteerd die ik niet zie in de query zelf

erikap commented on sparql-helpers/collaboration-activities.sparql.js:
```diff
@@ -0,0 +1,58 @@
+                                    prov:used           ?pressReleaseURI.
+                                    
+        OPTIONAL { ?collaborationActivityURI prov:startedAtTime ?startedAtTime }
+        OPTIONAL { ?collaborationActivityURI ext:currentEditor  ?editor }
```
Deze bestaat niet meer in het data model

erikap commented on app.js:
```diff
@@ -1,18 +1,60 @@
+        if (!collaborationActivity) {
+            return res.sendStatus(404);
+        }
+        // retrieve the press-release related to the collaborationActivity
```
Deze comment is niet wat de lijn code hieronder doet

erikap commented on sparql-helpers/press-release.sparql.js:
```diff
@@ -0,0 +1,33 @@
+    WHERE {
+        ${sparqlEscapeUri(pressReleaseURI)}     a                   fabio:PressRelease;
+                                                dct:creator         ?creatorURI.
+        ?creatorURI                             mu:uuid             ?creatorId. 
```
Heb je de `createorId` nodig. In de rest van de code lijkt enkel de URI gebruikt te worden.

MiVvlt committed:
added config to readme

erikap commented on sparql-helpers/collaboration-activities.sparql.js:
```diff
@@ -0,0 +1,58 @@
+export async function copyCollaborationActivityToTemporaryGraph(collaborationActivity, collaborators, tempGraphURI) {
+
+    const start = collaborationActivity.startedAtTime ? `prov:startedAtTime   ${sparqlEscapeDateTime(collaborationActivity.startedAtTime)};` : '';
+    const currentEditor = collaborationActivity.editor ? `ext:currentEditor   ${sparqlEscapeUri(collaborationActivity.editor)};` : '';
```
Deze bestaat niet meer in het data model

erikap commented on app.js:
```diff
@@ -1,18 +1,60 @@
+import { moveGraph, removeGraph } from './helpers/graph-helpers';
+
 
 app.post('/collaboration-activities/:id/share', async (req, res, next) => {
```
Kan je in dit endpoint logging toevoegen zodat je in de logs kan lezen van waar naar waar de data verdeeld wordt

erikap commented on app.js:
```diff
@@ -1,18 +1,60 @@
+        // remove temporary graph
+        await removeGraph(tempGraph);
 
         res.sendStatus(202);
```
Alle processing is hier voltooid, dus `204 No Content` is juister dan `202 Accepted`

erikap commented on app.js:
```diff
@@ -1,18 +1,60 @@
+
+        // create temporary copy
+        const tempGraph = `http://mu.semte.ch/graphs/tmp-data-share/${generateUuid()}`;
+        await copyCollaborationActivityToTemporaryGraph(collaborationActivity, collaborators, tempGraph);
```
Waarom een aparte method om collaboration-activity te kopiëren en zit dat niet mee vervat in de config zoals bvb attachments?

erikap commented on helpers/generic-helpers.js:
```diff
@@ -0,0 +1,91 @@
+        } else
+            console.log(`Don't know how to escape type ${type}. Will escape as a string.`);
+        return sparqlEscapeString(value);
+    };
```
Staat diezelfde methode nog eens gedupliceerd in deze file?

erikap commented on config.js:
```diff
@@ -0,0 +1,132 @@
+`;
+
+export const PRESS_RELEASE_PROPERTIES = {
+    'resourcePredicate': 'fabio:PressRelease',
```
Dit is geen predicate, maar een een type/class

erikap commented on config.js:
```diff
@@ -0,0 +1,132 @@
+            'http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#fileSize',
+            'http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#fileCreated',
+        ],
+        'relations': [
```
Je maakt het hier complex met die geneste relaties. Je kan dit generieke opvatten als je je lostrekt van het onderscheid properties/relaties. In een triplestore zijn alles triples.

Wat we basically willen is voor een aantal resource types, die via een link gerelateerd zijn aan een persbericht, een aantal predicates/objects kopieren. Je kan dat configureren in een flat-list met een item voor ieder resource type waarbij je bij ieder type specifieert:
- wat het path is naar het persbericht. Indien dit path meer dan 1 segment is (de 'nested' relations) kan je hier SPARQL property paths voor gebruiken
- welke properties je voor de resource wil kopieren

Als je dat voor elke resource doet, gaan de triples die de resources linken aan elkaar automatisch meegekopieerd worden.

De code om de temp-graph op te stellen is dan niet meer dan 1 for-loop over die config array en voor ieder resource type de insert data query genereren.

MiVvlt committed:
refactor copy press release

MiVvlt committed:
changed response status

MiVvlt committed:
fixes review and refactor copy collaboration activity

MiVvlt committed:
fixes

MiVvlt committed:
fixes and reamde

erikap committed:
Insert data per resource in tmp graph

erikap committed:
Simplify retrieval of properties for all configured resources

erikap committed:
Update resource configuration to share between organizations

- Add rdf:type for readability
- Add mu:uuid and rdf:type property on all resources
- Add press-release-activities, token-claims and approval-activities
- Remove resources already shared via public graph

erikap committed:
Prevent execution of empty insert data query

erikap committed:
Fix beleidsvelden predicate in config

erikap committed:
Add support to copy inverse properties of resources
